public/app/features/explore/TraceView/createSpanLink.tsx

```diff
@@ -332,21 +327,15 @@ function getLinkForSplunk(
  */
 function getTimeRangeFromSpan(
   span: TraceSpan,
-  timeShift: { startMs: number; endMs: number } = { startMs: 0, endMs: 0 },
-  isSplunkDS = false
+  timeShift: { startMs: number; endMs: number } = { startMs: 0, endMs: 0 }
 ): TimeRange {
   const adjustedStartTime = Math.floor(span.startTime / 1000 + timeShift.startMs);
   const from = dateTime(adjustedStartTime);
   const spanEndMs = (span.startTime + span.duration) / 1000;
   let adjustedEndTime = Math.floor(spanEndMs + timeShift.endMs);
 
-  // Splunk requires a time interval of >= 1s, rather than >=1ms like Loki timerange in below elseif block
-  if (isSplunkDS && adjustedEndTime - adjustedStartTime < 1000) {
+  if (adjustedEndTime - adjustedStartTime < 1000) {
```

Comment:
Yeah it is an issue for the customer that the logs are not perfectly aligned - that is why they are not seeing any and think the time range is not working properly.

I felt like adding just 1ms to the range was a bit small - and usually would result in no logs being found - so wanted to make it easier for the user so they would not be confused and without having to modify config settings.

It sounds like this is not really an issue if we are happy to leave it the way it is? I would close this PR and inform the customer that they should adjust the span start/end shift and explain why they are not seeing logs. What do you think?

Reply:
Spoke with Andrej about this in our standup. Will close this PR and link the customer to our docs on using the time shift + explain why they may not see logs instead of us changing the default.

Reply:
> I felt like adding just 1ms to the range was a bit small

btw we do this only in case the diff is less than <1ms. That would end up having range where `start === end` which would not work. Otherwise we just take start and end of the span time range.

At some point we had some default shift so we had `start - 1s` - `end + 1s` or something like that to widen the range but that ended up being too much buffer for lots of cases so we ended up making it configurable because it seems to be application specific.